partner_multi_relation/models/res_partner_relation_type.py

```diff
@@ -158,31 +158,63 @@ def get_category_condition(vals, side):
                                 relation.date_end > cutoff_date):
                             relation.write({'date_end': cutoff_date})
 
+    def _get_reflexive_relations(self):
+        """Get all reflexive relations for this relation type.
+
+        :return: a recordset of res.partner.relation.
+        """
+        self.env.cr.execute(
+            """
+            SELECT id FROM res_partner_relation
+            WHERE left_partner_id = right_partner_id
+            AND type_id = %(relation_type_id)s
+            """, {
+                'relation_type_id': self.id,
+            }
+        )
+        reflexive_relation_ids = [r[0] for r in self.env.cr.fetchall()]
+        return self.env['res.partner.relation'].browse(reflexive_relation_ids)
+
     def _check_no_existing_reflexive_relations(self):
-        """Check that no reflexive relation exists for this relation type."""
+        """Check that no reflexive relation exists for these relation types."""
         for relation_type in self:
-            self.env.cr.execute(
-                """
-                SELECT id FROM res_partner_relation
-                WHERE left_partner_id = right_partner_id
-                AND type_id = %(relation_type_id)s
-                """, {
-                    'relation_type_id': relation_type.id,
-                }
-            )
-            reflexive_relation_ids = [r[0] for r in self.env.cr.fetchall()]
-
-            if reflexive_relation_ids:
-                reflexive_relations = self.env['res.partner.relation'].browse(
-                    reflexive_relation_ids)
+            relations = relation_type._get_reflexive_relations()
+            if relations:
                 raise ValidationError(
                     _("Reflexivity could not be disabled for the relation "
                       "type {relation_type}. There are existing reflexive "
                       "relations defined for the following partners: "
                       "{partners}").format(
                         relation_type=relation_type.display_name,
-                        partners=reflexive_relations.mapped(
-                            'left_partner_id.display_name')))
+                        partners=relations.mapped('left_partner_id.display_name')))
+
+    def _delete_existing_reflexive_relations(self):
+        """Delete existing reflexive relations for these relation types."""
+        for relation_type in self:
+            relations = relation_type._get_reflexive_relations()
+            relations.unlink()
+
+    def _end_active_reflexive_relations(self):
```

Comment:
This will end all current reflexive relations that should be ended. However there might still be future reflexive relations, that is where date_start is after today. Admittedly, this will not occur often. But these can simply be deleted (as they will never come into effect). Compare what happens when relations are ended because of a change in category (from line 150, with ```# Delete future record...```.

Reply:
Might make a small utility method 'end_relations()' that will end the relation(s) if current, delete them when future: avoid code duplication.

Reply:
@NL66278 ok, I get it. Thanks.

Reply:
@NL66278 fixed